### PR TITLE
Updated fullyQualifiedName and LibraryUtilities.ts

### PR DESCRIPTION
**Pull request by yeexinc**
### Purpose

Added fullyQualifiedName property into NodeSearchElement.
The program should automatically generate an unformatted JSON file in the docs folder after running.

### CODE REVIEW

File: src/HostedContents/src/LibraryUtilities.ts [hunk 1]
```diff
@@ -11,7 +11,7 @@ class TypeListNode {
     memberType: MemberType = "none";
 
     constructor(data: any) {
-        this.fullyQualifiedName = data.creationName.split('@')[0]; // TODO: remove this hack
+        this.fullyQualifiedName = data.fullyQualifiedName;
```
Comment (benglin) [inline src/HostedContents/src/LibraryUtilities.ts:14]: Wow thanks for including this one! 👍

File: src/DynamoCore/Search/SearchElements/ZeroTouchSearchElement.cs [hunk 2]
```diff
@@ -23,6 +23,11 @@ public class ZeroTouchSearchElement : NodeSearchElement
         public override string CreationName { get { return functionDescriptor != null ? functionDescriptor.MangledName : this.Name; } }
 
         /// <summary>
+        /// The name that is used for Web UI
+        /// </summary>
+        public override string FullyQualifiedName { get { return functionDescriptor != null ? functionDescriptor.QualifiedName : this.Name; } }
```
Comment (benglin) [inline src/DynamoCore/Search/SearchElements/ZeroTouchSearchElement.cs:28]: Do we know what are the cases where `functionDescriptor` is `null`?

File: src/DynamoCore/Search/SearchElements/NodeSearchElement.cs [hunk 3]
```diff
@@ -51,6 +51,11 @@ public bool IsVisibleInSearch
         public virtual string CreationName { get { return this.Name; } }
 
         /// <summary>
+        /// The name that is used during Web UI (This is the same as FullName)
+        /// </summary>
+        public virtual string FullyQualifiedName { get { return FullCategoryName + "." + Name; } }
```
Comment (benglin) [inline src/DynamoCore/Search/SearchElements/NodeSearchElement.cs]: Can you make this `abstract` and make each derived class implement it?

File: src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs [hunk 4]
```diff
@@ -700,17 +700,28 @@ private void SetLoadedTypesRaw(ILibraryContainer libraryContainer, IEnumerable<N
...
+        private void WriteLibraryToFile(System.Text.StringBuilder builder)
+        {
+            // Write the unformatted JSON directly in the docs folder in HostedContents
+            string path = System.IO.Directory.GetCurrentDirectory() + "\\..\\..\\..\\src\\HostedContents\\docs\\";
+            using (StreamWriter sw = new StreamWriter(path + "RawTypeData.json"))
+            {
+                sw.WriteLine(builder.ToString());
+            }
+        }
+
```
Comment (benglin) [inline src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs:724]: Can you also include the generated (beautified) `RawTypeData.json` file here? I'll work on a Mac without Dynamo so I won't be able to run this. Thanks! :)

File: src/DynamoCore/Search/SearchElements/ZeroTouchSearchElement.cs (hunk 2 above)
Comment (yeexinc) [reply src/DynamoCore/Search/SearchElements/ZeroTouchSearchElement.cs:28]: I checked the nodes, so far I have yet to find a node that has `null` as its `functionDescriptor`.

**Comment by yeexinc** [comment]
Thanks @Benglin , updated the codes and added a `RawTypeData.json` file.

File: src/DynamoCore/Search/SearchElements/ZeroTouchSearchElement.cs (hunk 2 above)
Comment (benglin) [reply src/DynamoCore/Search/SearchElements/ZeroTouchSearchElement.cs:28]: Nice, thanks @yeexinc!

**Comment by benglin** [comment]
Thanks again, @yeexinc! Awesome job! 👍

**Pull request merged**